phpunit/html/wp-html-tag-processor-test.php

```diff
@@ -516,12 +516,159 @@ public function data_set_attribute_escapable_values() {
 	 *
 	 * @covers set_attribute
 	 * @covers get_updated_html
+	 * @covers get_attribute
 	 */
 	public function test_set_attribute_with_a_non_existing_attribute_adds_a_new_attribute_to_the_markup() {
 		$p = new WP_HTML_Tag_Processor( self::HTML_SIMPLE );
 		$p->next_tag();
 		$p->set_attribute( 'test-attribute', 'test-value' );
 		$this->assertSame( '<div test-attribute="test-value" id="first"><span id="second">Text</span></div>', $p->get_updated_html() );
+		$this->assertSame( 'test-value', $p->get_attribute( 'test-attribute' ) );
+	}
+
+	/**
+	 * @ticket 56299
+	 *
+	 * @covers set_attribute
+	 * @covers get_updated_html
+	 * @covers get_attribute
+	 */
+	public function test_get_attribute_returns_updated_values_before_they_are_updated() {
+		$p = new WP_HTML_Tag_Processor( self::HTML_SIMPLE );
+		$p->next_tag();
+		$p->set_attribute( 'test-attribute', 'test-value' );
+		$this->assertSame( 'test-value', $p->get_attribute( 'test-attribute' ) );
```

Comment:
The convention in WordPress core, where these tests will eventually be merged, is to provide an error message every time a test fails. This means adjusting all the new `assert` calls as follows:

```suggestion
		$this->assertSame( 'test-value', $p->get_attribute( 'test-attribute' ), "get_attribute() did not return "test-value" for the attribute "test-attribute" after "test-value" was assigned using set_attribute()." );
```

I know 😬